ui/components/src/TimeChart/TimeChart.tsx

```diff
@@ -254,13 +267,38 @@ export const TimeChart = forwardRef<ChartInstance, TimeChartProps>(function Time
     __experimentalEChartsOptionsOverride,
     noDataVariant,
     timeZone,
+    tooltipPinnedCoords,
     isStackedBar,
   ]);
 
   return (
     <Box
       sx={{ height }}
       onClick={(e) => {
```

Comment:
Another option could be to use approach here: https://apache.github.io/echarts-handbook/en/concepts/event/#listen-to-events-on-the-blank-area

Using regular `onClick` the way we've been doing it seems like it's more straightforward for React devs than using `chart.getZr().on('click', (e) => {});` but open to feedback or other ideas. 

There are also these out-of-the-box [events](https://echarts.apache.org/en/api.html#events), but they mostly trigger when interacting with lines / datapoint symbols instead of clicking anywhere on the canvas like we need for pinned tooltip behavior

Reply:
wonder if we can reuse the logic in the tooltip to find the nearest data point and snap the pinned crosshair there?

Reply:
Would adding back snapping help with that though?

Reply:
i don't think so because i think this is what we want ideally:

1. while hovering the graph, the crosshair doesn't snap but the nearest data points are highlighted
2. the crosshair only snaps to the nearest data point when user pins the tooltip 

does this sound right?